src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs

```diff
@@ -357,6 +357,51 @@ await foreach (var streamValue in asyncEnumerable)
             }
         }
 
+        [Theory]
+        [MemberData(nameof(HubProtocolsAndTransportsAndHubPaths))]
```

Comment:
Why would this test be affected by hub paths? This is 15 variations! 2 hub protocols * 3 transport types * 3 paths - 3 variations since MessagePack and SSE are incompatible no matter the path.

And then to compound the problem, each variation waits at least 50ms in Task.Delays on top of the time needed to start and stop the server, start and stop the connection, etc...

Generally speaking, avoid variations in tests unless the variations really are interesting. If you're using the same MemberData for a bunch of different tests, that's usually a sign of way too many variations.

Reply:
I will agree this is a slippery slope particularly because of the delays in this test. I changed it to just vary over protocols, fixing the hub path at default and transport to websockets (was considering varying over all transports as well though.)